cve_bin_tool/nvd_api.py

```diff
@@ -0,0 +1,277 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Retrieval access and of NVD entries using NVD Automatic CVE Retrieval
+"""
+
+import asyncio
+import gzip
+import json
+import math
+import os
+import time
+from datetime import datetime
+from logging import Logger
+from typing import Union
+
+import aiohttp
+from rich.progress import Progress, track
+
+from cve_bin_tool.async_utils import FileIO, GzipFile, RateLimiter, run_coroutine
+from cve_bin_tool.error_handler import ErrorMode
+from cve_bin_tool.log import LOGGER
+
+NVD_FILENAME_TEMPLATE = "nvdcve-1.1-{}.json.gz"
+DISK_LOCATION_DEFAULT = os.path.join(os.path.expanduser("~"), ".cache", "cve-bin-tool")
+
+
+FEED = "https://services.nvd.nist.gov/rest/json/cves/1.0"
+
+PAGESIZE = 2000
+MAX_FAIL = 5
+# Interval in seconds between successive requests
+INTERVAL_PERIOD = 2
+
+
+def get_utc_offset():
+    """Returns the UTC offset string. Example: `UTC-00:00`,`UTC-05:30`  """
+    minute = (time.localtime().tm_gmtoff / 60) % 60
+    hour = ((time.localtime().tm_gmtoff / 60) - minute) / 60
+    return f"UTC-{int(hour):02d}:{int(minute):02d}"
+
+
+def filter_by_id(cve, update_data):
+    """"Function to filter out duplicate CVE entries in case of incremental update"""
+    for update_cve in update_data:
+        if (
+            update_cve["cve"]["CVE_data_meta"]["ID"]
+            == cve["cve"]["CVE_data_meta"]["ID"]
+        ):
+            return False
+    return True
```

Comment:
```suggestion
def filter_by_id(cve, update_data):
    """Function to filter out duplicate CVE entries in case of incremental update"""
    id = cve["cve"]["CVE_data_meta"]["ID"]
    return all(
        id != update_cve["cve"]["CVE_data_meta"]["ID"] for update_cve in update_data
    )
```
This reads easier imo.